### PR TITLE
Replace and remove dependencies with lighter and audited alternatives

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [ ] **If this is a code change**: I have written unit tests.
- [x] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Reuse noble-hashes instead of js-sha256 to reduce the amount of dependencies.
Reuse noble-curves instead of elliptic to reduce the amount of dependencies.
Remove error-polyfill package since it is no longer used explicitly.

## Test Plan

All tests pass

## Related issues/PRs

https://github.com/near/near-api-js/pull/1217
https://github.com/near/near-api-js/issues/1200

### CODE REVIEW

File: packages/signers/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   "dependencies": {
     "@near-js/crypto": "workspace:*",
     "@near-js/keystores": "workspace:*",
-    "js-sha256": "0.9.0"
+    "@noble/hashes": "^1.3.3"
```
Comment (vikinatora) [inline packages/signers/package.json:20]: Please lock the version

File: packages/transactions/package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "@near-js/utils": "workspace:*",
     "bn.js": "5.2.1",
     "borsh": "1.0.0",
-    "js-sha256": "0.9.0"
+    "@noble/hashes": "^1.3.3"
```
Comment (vikinatora) [inline packages/transactions/package.json:25]: Same comment

File: packages/near-api-js/src/browser-index.ts [hunk 3]
```diff
@@ -2,5 +2,3 @@
 export * as keyStores from './key_stores/browser-index';
 export * from './common-index';
 export * from './browser-connect';
-
-import 'error-polyfill';
```
Comment (vikinatora) [inline packages/near-api-js/src/browser-index.ts:6]: I think it's fine to leave error-polyfill

**Review by vikinatora — CHANGES_REQUESTED** [review]

File: packages/near-api-js/src/browser-index.ts (hunk 3 above)
Comment (vikinatora) [reply packages/near-api-js/src/browser-index.ts:6]: Disregard the comment

**Review by vikinatora — DISMISSED** [review]

**Review by gagdiez — APPROVED** [review]

**Review by vikinatora — APPROVED** [review]

**Pull request merged**